server/lib/cloudflare.js

```diff
@@ -0,0 +1,44 @@
+import config from 'config';
+import cloudflare from 'cloudflare';
+
+import logger from './logger';
+
+// Load config
+const cfConfig = config.cloudflare || {};
+
+// Check config
+const isLiveServer = ['staging', 'production'].includes(config.env);
+const hasConfig = Boolean(cfConfig.email && cfConfig.key && cfConfig.zone);
+
+if (isLiveServer && !hasConfig) {
+  logger.warn('Your Cloudflare config is imcomplete, you must provide an email, a key and a zone');
+} else if (!isLiveServer && hasConfig) {
+  logger.info('A Cloudflare config was provided on a Live/Test environment. Some methods will be stubbed.');
+}
+
+const CloudflareLib = cloudflare({ email: cfConfig.email, key: cfConfig.key });
+
+// Export some helpers
+
+/**
+ * Purge the given page from cloudflare's cache. In dev environments this function
+ * will only log to the console and will skip the call to cloudflare.
+ *
+ * @param {string|array} pagePaths - a path or an array of paths `/eslint`, `['/', '/about']`
+ * @returns {Promise}
+ */
+export const purgeCacheForPage = pagePaths => {
+  const prepareURL = pagePath => {
+    const prependSlash = pagePath[0] === '/' ? '' : '/';
+    return `${config.host.website}${prependSlash}${pagePath}`;
```

Comment:
Have you tested?

My experience of Cloudflare is that we need to empty `https://frontend.opencollective.com/babel` not `https://opencollective.com/babel`. I personally have the habit of purging both, not sure it's justified.

Reply:
I tested a bit naively, just ensured the collective page was properly updated.

I just did more advanced testing with `curl -svo /dev/null` to get the headers and it seems like I was lucky during my tests; the cache is not purged properly.

However I don't have more luck by doing:

```es6
api.zones.purgeCache('...', {files: ['https://staging.opencollective.com/captainfact_io', 'https://staging-frontend.opencollective.com/captainfact_io']})
```

Reply:
It's `https://frontend-staging.opencollective.com`.